### PR TITLE
BLD: ignore multiple types of file in wheel

### PR DESCRIPTION
**Pull request by fangchenli**
xref #40169

### CODE REVIEW

**Comment by jreback** [comment]
lgtm

@simonjayhawkins give this a try and merge / backport if good

**Review by jreback — APPROVED** [review]

**Comment by jbrockmendel** [comment]
thanks @fangchenli

**Comment by simonjayhawkins** [comment]
Thanks @fangchenli 

will open PR on pandas-wheels to test

**Comment by simonjayhawkins** [comment]
> will open PR on pandas-wheels to test

https://github.com/MacPython/pandas-wheels/pull/135

**Comment by simonjayhawkins** [comment]
we still need the pxi.in template files

```
    File "/tmp/pip-build-env-jawd5f8v/overlay/lib/python3.7/site-packages/setuptools/build_meta.py", line 135, in _get_build_requires
      self.run_setup()
    File "/tmp/pip-build-env-jawd5f8v/overlay/lib/python3.7/site-packages/setuptools/build_meta.py", line 259, in run_setup
      self).run_setup(setup_script=setup_script)
    File "/tmp/pip-build-env-jawd5f8v/overlay/lib/python3.7/site-packages/setuptools/build_meta.py", line 150, in run_setup
      exec(compile(code, __file__, 'exec'), locals())
    File "setup.py", line 650, in <module>
      ext_modules=maybe_cythonize(extensions, compiler_directives=directives),
    File "setup.py", line 422, in maybe_cythonize
      build_ext.render_templates(_pxifiles)
    File "setup.py", line 91, in render_templates
      with open(pxifile) as f:
  FileNotFoundError: [Errno 2] No such file or directory: 'pandas/_libs/algos_common_helper.pxi.in'
```

**Comment by simonjayhawkins** [comment]
The wheels on pandas-wheels look to be building OK, these must be built from source. the failure in previous comment is from https://github.com/simonjayhawkins/pandas-release/actions/runs/932775977

File: MANIFEST.in [hunk 1]
```diff
@@ -39,6 +40,14 @@ global-exclude .DS_Store
 global-exclude .git*
 global-exclude \#*
 
+global-exclude *.c
+global-exclude *.cpp
+global-exclude *.h
+
+global-exclude *.py[codx]
```
Comment (simonjayhawkins) [inline MANIFEST.in:47]: i think this should be `*.py[ocd]` to match .gitignore

File: MANIFEST.in [hunk 2]
```diff
@@ -39,6 +40,14 @@ global-exclude .DS_Store
...
+global-exclude *.px[di]
```
Comment (simonjayhawkins) [inline MANIFEST.in:48]: and `*.pxi`

File: MANIFEST.in [hunk 3]
```diff
@@ -39,6 +40,14 @@ global-exclude .DS_Store
...
+global-exclude *.pxi.in
```
Comment (simonjayhawkins) [inline MANIFEST.in:49]: and remove this.

will submit a PR once get pip-test and conda-test to pass locally (built from sdist) with these changes

File: MANIFEST.in [hunk 4]
```diff
@@ -39,6 +40,14 @@ global-exclude .DS_Store
 global-exclude .git*
 global-exclude \#*
 
+global-exclude *.c
+global-exclude *.cpp
+global-exclude *.h
```
Comment (simonjayhawkins) [inline MANIFEST.in:45]: We need to keep the  .c and .h for vendored code e.g. parsers etc

Comment (simonjayhawkins) [reply MANIFEST.in:45]: `graft pandas/_libs/src` seems to work ok (and need to do the same for np_datetime.c and another couple of files)

**Pull request merged**